modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/service/TemplateProcessorComponent.java

```diff
@@ -173,58 +138,44 @@ public void addContextExtensions(DocumentModel currentDocument, DocumentWrapper
     @Override
     public List<String> getReservedContextKeywords() {
         List<String> keywords = new ArrayList<>();
-        Map<String, ContextExtensionFactoryDescriptor> factories = contextExtensionRegistry.getExtensionFactories();
-        for (String name : factories.keySet()) {
-            keywords.add(name);
-            keywords.addAll(factories.get(name).getAliases());
-        }
-        for (String keyword : AbstractContextBuilder.RESERVED_VAR_NAMES) {
-            keywords.add(keyword);
+        List<ContextExtensionFactoryDescriptor> contribs = getRegistryContributions(CONTEXT_EXTENSION_XP);
+        for (ContextExtensionFactoryDescriptor factoryDesc : contribs) {
+            keywords.add(factoryDesc.getName());
+            keywords.addAll(factoryDesc.getAliases());
         }
+        CollectionUtils.addAll(keywords, AbstractContextBuilder.RESERVED_VAR_NAMES);
         return keywords;
     }
 
     @Override
     public Map<String, ContextExtensionFactoryDescriptor> getRegistredContextExtensions() {
-        return contextExtensionRegistry.getExtensionFactories();
+        Map<String, ContextExtensionFactoryDescriptor> map = this.<MapRegistry> getExtensionPointRegistry(
+                CONTEXT_EXTENSION_XP).getContributions();
+        return Collections.unmodifiableMap(map);
     }
 
-    protected TemplateProcessorDescriptor findProcessorByMimeType(String mt) {
-        List<TemplateProcessorDescriptor> candidates = new ArrayList<>();
-        for (TemplateProcessorDescriptor desc : processorRegistry.getRegistredProcessors()) {
-            if (desc.getSupportedMimeTypes().contains(mt)) {
-                if (desc.isDefaultProcessor()) {
-                    return desc;
-                } else {
-                    candidates.add(desc);
-                }
-            }
+    protected Optional<TemplateProcessorDescriptor> findProcessor(Predicate<TemplateProcessorDescriptor> predicate) {
+        List<TemplateProcessorDescriptor> candidates = this.<TemplateProcessorDescriptor> getRegistryContributions(
+                PROCESSOR_XP).stream().filter(predicate).collect(Collectors.toList());
+        Optional<TemplateProcessorDescriptor> res = candidates.stream()
+                                                              .filter(TemplateProcessorDescriptor::isDefaultProcessor)
+                                                              .findFirst();
+        if (res.isEmpty() && !candidates.isEmpty()) {
+            res = Optional.of(candidates.get(0));
         }
-        if (candidates.size() > 0) {
-            return candidates.get(0);
-        }
-        return null;
+        return res;
     }
```

Comment:
What do think about sorting the stream just after giving the `predicate`, the boolean natural order is _false_ then _true_, we will end with something like:
```
return this.<TemplateProcessorDescriptor> getRegistryContributions(PROCESSOR_XP)
           .stream()
           .filter(predicate)
           .sorted(Comparator.comparing(TemplateProcessorDescriptor::isDefaultProcessor).reversed()) // true first
           .findFirst();
```

Reply:
Thanks, i wondered how to even more simplify this. The only potential difference is that original order of contributions is not respected, i don't know if this should taken into account.

Reply:
Let's see how tests behave with this change: maybe this edge case should not happen in practice (?)

Reply:
As far as I know, the sort won't change the original order of contributions having the same value for `isDefaultProcessor` 🤔. That's an interesting interrogation, not sure we need to respect it as we don't (yet) provide any ordering mechanism. Wdyt?

Reply:
Well we do provide some kind of ordering mechanism, we could imagine a component requiring the other just to make sure its contributions are registered after.
Anyway this would not change the fact that first is taken in any case here, so if the sort does not change the original order for those, we should be good.